src/Raven.Server/Config/ConfigurationEntryValue.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using Raven.Server.Config.Attributes;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Config
+{
+    public abstract class ConfigurationEntryValue : IDynamicJson
+    {
+        protected ConfigurationEntryValue(ConfigurationEntryMetadata metadata)
+        {
+            Metadata = metadata ?? throw new ArgumentNullException(nameof(metadata));
+        }
+
+        public ConfigurationEntryMetadata Metadata { get; }
+
+        public virtual DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(Metadata)] = Metadata.ToJson()
+            };
+        }
+    }
+
+    public class ConfigurationEntryServerValue : ConfigurationEntryValue
+    {
+        public ConfigurationEntryServerValue(RavenConfiguration configuration, ConfigurationEntryMetadata metadata, RavenServer.AuthenticationStatus authenticationStatus)
+            : base(metadata)
+        {
+            ServerValues = new Dictionary<string, ConfigurationEntrySingleValue>();
+            foreach (var key in Metadata.Keys)
+            {
+                var value = configuration.GetServerWideSetting(key);
+                if (value == null)
+                    continue;
+
+                var hasAccess = authenticationStatus == RavenServer.AuthenticationStatus.ClusterAdmin;
+                ServerValues[key] = new ConfigurationEntrySingleValue
+                {
+                    Value = hasAccess ? value : null,
+                    HasAccess = hasAccess
```

Comment:
Where is this being used?

Reply:
It is for the UI, so we will be able to write there that the value is set, but user do not have access to it (e.g. DatabaseAdmin should not see ServerWideOnly values)

Reply:
Where are you doing the filtering?

Reply:
There was no intention to do any filtering. It depends how we will design the UI.

Reply:
Let me rephrase, where are you replacing the value with `****`

Reply:
Done